runtime/interpreter/interpreter.go

```diff
@@ -5474,3 +5468,116 @@ func (interpreter *Interpreter) withResourceDestruction(
 
 	f()
 }
+
+// Converts the input value into a version compatible with the new entitlements feature,
+// with the same members/operations accessible on any references as would have been accessible in the past.
+// Modifies the input `v` in place.
+// This is used for migrations, but must be located in this package because some of the fields we wish to modify are
+// privately scoped to this package
+func (interpreter *Interpreter) ConvertValueToEntitlements(
```

Comment:
This defines the main logic of the migration: converting a value from before entitlements into a post-entitlements value

Reply:
Can this be moved to `migrations/entitlements/migration.go`?

Reply:
Unfortunately it uses some fields that are privately scoped to the interpreter package

Reply:
This can be moved now with later changes!